src/coord/src/catalog.rs

```diff
@@ -1013,6 +1013,48 @@ impl Catalog {
         }
     }
 
+    pub fn custom_oid_dependencies(&self, item: &CatalogItem) -> Option<Vec<u32>> {
+        match item {
+            CatalogItem::Table(Table { desc, .. }) | CatalogItem::Source(Source { desc, .. }) => {
+                Some(desc.typ().custom_type_oids())
+            }
+            CatalogItem::View(View { optimized_expr, .. }) => {
```

Comment:
I am afraid the check does not recurse into the expressions correctly. This ```DROP TYPE``` is allowed to succeed and borks the catalog:

```
materialize=> CREATE TYPE int4_list AS LIST (element_type=int4)
materialize-> ;
CREATE TYPE
materialize=> CREATE VIEW v1 AS SELECT CAST(CAST('{2}' AS int4_list) AS text); 
CREATE VIEW
materialize=> drop type int4_list;
DROP TYPE
```

This one too:

```
materialize=> CREATE VIEW v1 AS SELECT MIN(CAST(CAST('{1}' AS int4_list) AS string));
CREATE VIEW
materialize=> drop type int4_list;
DROP TYPE
```